grammar.js

```diff
@@ -519,10 +519,10 @@ module.exports = grammar({
       '(',
       choice(
         field('initializer', $.declaration),
-        seq(field('initializer', optional($._expression)), ';')
+        seq(field('initializer', commaSep($._expression)), ';')
       ),
       field('condition', optional($._expression)), ';',
-      field('update', optional(choice($._expression, $.comma_expression))),
+      field('update', commaSep($._expression)),
```

Comment:
Did `comma_expression` not work here? I would have thought it would work. I would also think that it would work for the initializer.

Reply:
You're right: I misread the recursive comma_expression.
And I thought it was better to have a "list" of initiliazers but when dumping the ast from clang, ',' is really a bin operator.
```bash
echo 'void f() {
    int i, j, k, l;
    for (i = 0, j = 0, k = 0, l = 0; i < 1; i++, j++, k++, l++) { }
}' | clang-9 -Xclang -ast-dump -x c -c -
```
So I'll update my patch.